projects/media-viewer/src/lib/annotations/annotation.component.html

```diff
@@ -1,21 +1,22 @@
-<div class="annotation-wrapper" tabindex="1" (focusout)="toggleSelection(false)">
+<ng-container *ngFor="let rectangle of annotation.rectangles; index as i">
   <mv-anno-rectangle
     (click)="toggleSelection(true)"
-    (update)="onRectangleUpdate($event)"
     [selected]="selected"
     [zoom]="zoom"
-    [rectangle]="annotation.rectangles[0]"
+    [rectangle]="rectangle"
     [draggable]="draggable"
     [color]="annotation.color">
   </mv-anno-rectangle>
-  <mv-anno-comment
-    *ngIf="annotation.comments[0]"
-    (click)="toggleSelection(true)"
-    (delete)="deleteComment()"
-    (updated)="onCommentUpdate($event)"
-    [selected]="selected"
-    [comment]="annotation.comments[0]"
-    [left]="commentsLeftOffset"
-    [top]="annotation.rectangles[0].y *(zoom)">
-  </mv-anno-comment>
-</div>
+</ng-container>
+
+<mv-anno-comment
+  *ngIf="annotation.comments[0]"
+  (click)="toggleSelection(true)"
+  [rotate]="rotate"
+  (delete)="deleteComment()"
+  (updated)="onCommentUpdate($event)"
+  [selected]="selected"
+  [comment]="annotation.comments[0]"
+  [rectangle]="annotation.rectangles[0]"
```

Comment:
this doesn't seem right, I assume we're passing the rectangle to get the "top" value, so we can't use the same rectangle for all comments. not sure how comments are sorted relative to rectangles

Reply:
top, left, width and height - it's needed for rotation